### PR TITLE
Add note about num.standby.replicas

### PR DESCRIPTION
**Pull request by JimGalasyn**

### CODE REVIEW

**Comment by JimGalasyn** [comment]
@guozhangwang can you please merge? Thanks!

File: docs/streams/developer-guide/config-streams.html [hunk 1]
```diff
@@ -418,12 +418,17 @@ <h4><a class="toc-backref" href="#id9">default.value.serde</a><a class="headerli
...
+              specified number of replicas per store and keep them up to date as long as there are enough instances running.
               Standby replicas are used to minimize the latency of task failover.  A task that was previously running on a failed instance is
               preferred to restart on an instance that has standby replicas so that the local state store restoration process from its
               changelog can be minimized.  Details about how Kafka Streams makes use of the standby replicas to minimize the cost of
               resuming tasks on failover can be found in the <a class="reference internal" href="../architecture.html#streams_architecture_state"><span class="std std-ref">State</span></a> section.</div></blockquote>
-        </div>
+            </div>
+            <div class="admonition note">
+              <p class="first admonition-title">Note</p>
+              <p class="last">If you enable <cite>n</cite> standby tasks, you need to provision <cite>n+1</cite> <code class="docutils literal"><span class="pre">KafkaStreams</span></code>
```
Comment (guozhangwang) [inline docs/streams/developer-guide/config-streams.html:429]: This is not true... you can still have just one instance, which will host all n+1 tasks. You'd probably say `provision for n+1 replicas for each state stores`.

Comment (JimGalasyn) [reply docs/streams/developer-guide/config-streams.html:429]: @mjsax , can you please weigh in?

Comment (mjsax) [reply docs/streams/developer-guide/config-streams.html:429]: @guozhangwang From my understanding, we would never assign a StandbyTask to the same KafkaStreams instance that hosts the corresponding active tasks. Thus, for replication N, you need N+1 KafkaStreams instances. Or do I miss something?

Comment (guozhangwang) [reply docs/streams/developer-guide/config-streams.html:429]: Ack, that is right. I just confirmed from `StickyTaskAssignor` that if we do not have enough instances we will see a WARN log and not enough standby tasks as required will be assigned.

Comment (JimGalasyn) [reply docs/streams/developer-guide/config-streams.html:429]: @guozhangwang , is this okay to merge?

Comment (guozhangwang) [reply docs/streams/developer-guide/config-streams.html:429]: Yes, done :)

Comment (JimGalasyn) [reply docs/streams/developer-guide/config-streams.html:429]: Great, thanks!

**Pull request merged**